Core/Core/Views/HorizontalMenuView.swift

```diff
@@ -0,0 +1,132 @@
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell: HorizontalMenuViewCell = collectionView.dequeue(for: indexPath)
+        cell.title = delegate?.menuItemTitle(at: indexPath)
+        cell.selectionColor = delegate?.selectedColor
+        cell.font = delegate?.measurementFont
+        if indexPath == selectedIndexPath {
+            cell.isSelected = true
+        }
+        return cell
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if let selectedIndexPath = selectedIndexPath {
+            let cell = collectionView.cellForItem(at: selectedIndexPath)
+            cell?.isSelected = false
+        }
+        selectedIndexPath = indexPath
+        delegate?.didSelectItem(at: indexPath)
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return sizeForItem(for: indexPath)
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        let spacing = spaceBetweenCells()
+        return UIEdgeInsets(top: 0, left: spacing, bottom: 0, right: spacing)
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumInteritemSpacingForSectionAt section: Int) -> CGFloat {
+        return spaceBetweenCells()
+    }
+
+    func sizeForItem(for indexPath: IndexPath) -> CGSize {
+        guard let font = delegate?.measurementFont,
+            let title = delegate?.menuItemTitle(at: indexPath),
+            let maxWidth = delegate?.maxItemWidth
+            else { return CGSize.zero }
+        let maxHeight: CGFloat = 40.0
+        let constraintRect = CGSize(width: maxWidth, height: maxHeight)
+        let boundingBox = title.boundingRect(with: constraintRect, options: [], attributes: [NSAttributedString.Key.font: font], context: nil)
+        return CGSize(width: ceil(boundingBox.width), height: maxHeight)
+    }
+
+    func spaceBetweenCells() -> CGFloat {
+        if let cachedSpaceBetweenCells = cachedSpaceBetweenCells { return cachedSpaceBetweenCells }
+
+        guard let itemCount = delegate?.menuItemCount() else { return 0.0 }
+        if cachedTotalWidthOfItems == nil {
+            var totalWidth: CGFloat = 0
+            for i in 0..<itemCount {
+                let size = sizeForItem(for: IndexPath(item: i, section: 0))
+                totalWidth += size.width
+            }
+            cachedTotalWidthOfItems = totalWidth
+        }
+        let space = floor( (frame.size.width - (cachedTotalWidthOfItems ?? 0)) / CGFloat(itemCount + 1) )
+        cachedSpaceBetweenCells = space
+        return space
+    }
+}
```

Comment:
I haven't done much with collection views yet. Is all this manual sizing of things necessary?

Reply:
this is kind of the magic centering all the horizontal menu items